cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -711,7 +713,9 @@ private SearchResults searchByDefaultIndex(SearchRequest request) {
         while ((next = scanner.next()) != null) {
           Optional<MetadataEntry> metadataEntry = parseRow(next, column, request.getTypes(),
                                                            request.shouldShowHidden());
-          metadataEntry.ifPresent(results::add);
+          if (metadataEntry.isPresent()) {
```

Comment:
It is better to use the `Optional` map then using `if` check:

```java
metadataEntry.map(e -> new MetadataResultEntry(e, searchTerm.getTerm())).ifPresent(results::add)
```

or if you like, you can do everything in one statement:

```java
          parseRow(next, column, request.getTypes(), request.shouldShowHidden())
            .map(e -> new MetadataResultEntry(e, searchTerm.getTerm()))
            .ifPresent(results::add);
```

Reply:
implemented stream in new commit